### PR TITLE
fix: add extra check before first DB update

### PR DESCRIPTION
**Pull request by pbalogh-sa**

### CODE REVIEW

File: grype-server/pkg/scanner/scanner.go [hunk 1]
```diff
@@ -175,6 +175,11 @@ func (s *Scanner) loadBb(cfg *db.Config) error {
 	if err != nil {
 		return fmt.Errorf("failed to create curator: %v", err)
 	}
+	// Check IsUpdateAvailable() errors because if it has errors, the update process will be continued during Update().
+	// https://github.com/anchore/grype/blob/731abaab723ae8918635d4e20399ca3c00b665f4/grype/db/curator.go#L138-L143
+	if _, _, err := dbCurator.IsUpdateAvailable(); err != nil {
```
Comment (akpsgit) [inline grype-server/pkg/scanner/scanner.go:180]: let's try to corrupt the DB on purpose to simulate a crashed OOM flow, and then we can start the grype server locally to see if it goes through this flow, and whether we can recover from a corrupted DB situation somehow.

Comment (pbalogh-sa) [reply grype-server/pkg/scanner/scanner.go:179]: ack

File: grype-server/pkg/config/config.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ const (
 	HealthCheckAddress = "HEALTH_CHECK_ADDRESS"
 	DbRootDir          = "DB_ROOT_DIR"
 	DbUpdateURL        = "DB_UPDATE_URL"
+	SqliteVersion      = "SQLITE_VERSION"
```
Comment (akpsgit) [inline grype-server/pkg/config/config.go:13]: Maybe better to call it DB_DIR?

File: grype-server/pkg/config/config.go [hunk 3]
```diff
@@ -18,6 +19,7 @@ type Config struct {
 	HealthCheckAddress string
 	DbRootDir          string
 	DbUpdateURL        string
+	SqliteVersion      string
```
Comment (akpsgit) [inline grype-server/pkg/config/config.go:22]: Maybe better to call it DbDir?

File: grype-server/pkg/config/config.go (hunk 2 above)
Comment (pbalogh-sa) [reply grype-server/pkg/config/config.go:13]: ack

**Review by akpsgit — APPROVED** [review]

**Pull request merged**